### PR TITLE
Add CycleGAN.

### PR DESCRIPTION
**Pull request by qingqing01**

### CODE REVIEW

File: cyclegan/data.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            if p > 0.5:
+                random_id = random.randint(0, self.pool_size - 1)
+                temp = self.pool[random_id]
+                self.pool[random_id] = image
+                return temp
+            else:
+                return image
+
+
+if __name__ == '__main__':
```
Comment (LielinJiang) [inline cyclegan/data.py:124]: delete?

File: cyclegan/infer.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
+# Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
```
Comment (LielinJiang) [inline cyclegan/infer.py:1]: 2020

File: cyclegan/data.py (hunk 1 above)
Comment (qingqing01) [reply cyclegan/data.py]: done

File: cyclegan/infer.py (hunk 2 above)
Comment (qingqing01) [reply cyclegan/infer.py:1]: done

File: cyclegan/train.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
+# Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
```
Comment (heavengate) [inline cyclegan/train.py:1]: 2020

File: cyclegan/train.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+                      format(epoch, i, g_loss[0], da_loss[0], db_loss[0], t))
+        g.save('{}/{}'.format(FLAGS.checkpoint_path, epoch))
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser("CycleGAN Training on Cityscapes")
+    parser.add_argument(
+        "-d", "--dynamic", action='store_false', help="Enable dygraph mode")
+    parser.add_argument(
+        "--device", type=str, default='gpu', help="device to use, gpu or cpu")
```
Comment (heavengate) [inline cyclegan/train.py:139]: add `-p`，which is same as `infer.py` ？

File: cyclegan/train.py [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+    g.prepare(g_optimizer, GLoss(), inputs=[input_A, input_B])
+    d_A.prepare(da_optimizer, DLoss(), inputs=[input_B, fake_B])
+    d_B.prepare(db_optimizer, DLoss(), inputs=[input_A, fake_A])
+
+    if FLAGS.resume:
+        g.load(FLAGS.resume)
+
+    loader_A = fluid.io.DataLoader(
+        data.DataA(),
+        feed_list=[x.forward() for x in [input_A]]
```
Comment (heavengate) [inline cyclegan/train.py:91]: no need to set `feed_list` in newest `DataLoader` code

File: cyclegan/train.py (hunk 3 above)
Comment (qingqing01) [reply cyclegan/train.py]: Done

File: cyclegan/train.py (hunk 5 above)
Comment (qingqing01) [reply cyclegan/train.py]: Done

File: cyclegan/train.py (hunk 4 above)
Comment (qingqing01) [reply cyclegan/train.py]: Done

**Pull request merged**